### ISSUE TITLE
SPSearchServiceApp is Creating a New SA Even with Existing DB

### ISSUE THREAD

**Issue by shurick81** [dsccommunity/SharePointDsc#1233]
#### Details of the scenario you tried and the problem that is occurring

Trying to restore the search service application from existing database. For example, in DR recovery scenario or blue-green deployments or stage infrastructure or for v2v upgrade scenarios.

#### Verbose logs showing the problem

Current version of the SharePointDsc module is using `New-SPEnterpriseSearchServiceApplication`, whereas when we need to restore a search service application from existing administration database, we need to use `Restore-SPEnterpriseSearchServiceApplication` commandlet.

#### Suggested solution to the issue

Check if the administration database already exists in the SQL instance and run either `New-SPEnterpriseSearchServiceApplication` or `Restore-SPEnterpriseSearchServiceApplication` depending on the condition.

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)

```powershell
# just an example,
        SPSearchServiceApp SearchServiceApp
        {
            DatabaseName                = "Search_Service_Application_DB_b183238874e14d7dad0aed1941f80308"
            ApplicationPool             = "SharePoint Search App Pool"
            ProxyName                   = "Search Service Application"
            Name                        = "Search Service Application"
            DefaultContentAccessAccount = $SPSearchCrawlAccountCredential
            Ensure                      = "Present"
            PsDscRunAsCredential        = $SPInstallAccountCredential
            DependsOn                   = "[Script]SearchServiceApp"
        }
```

**Comment by ykuijs** [comment]
Good point. SPFarm is using some code to check if the ConfigDB exists. We can reused that code here.

**Comment by ykuijs** [comment]
Documenting info:
[New-SPEnterpriseSearchServiceApplication](https://docs.microsoft.com/en-us/powershell/module/sharepoint-server/new-spenterprisesearchserviceapplication)
[Restore-SPEnterpriseSearchServiceApplication](https://docs.microsoft.com/en-us/powershell/module/sharepoint-server/restore-spenterprisesearchserviceapplication)

**Comment by ykuijs** [comment]
Hi @shurick81, I have been investigating this change some more. What is the exact scenario you are trying to accomplish?

The documentation for the Restore cmdlet, I see this:

> The first set of parameters is for Application Configuration Attach mode and the second set of parameters is for Search Application Attach mode.
> 
> Application Configuration Attach mode only restores configuration data that is stored in the administration database. However, Search Application Attach restores configuration, topology and all crawled data.

Which scenario are you looking for?

**Comment by shurick81** [comment]
Let me put it like this. I have a Disaster Recovery farm precreated. Then when necessary, I want to restore the copy of the administration database from production to the SQL instance of the DR farm and then run a DSC configuration that will attach it. 

According to my test, using `New-SPEnterpriseSearchServiceApplication` results in missing all my custom content sources in the service application. So I concluded that entire restored database was replaced with a fresh, empty database.
